src/server/index.js

```diff
@@ -46,9 +48,8 @@ app.use(morgan('dev', {
 app.use(cookieParser());
 app.use(express.static(path.join(__dirname, '../..', 'public')));
 
-app.use('/', require('./routes/databaseRoutes/sockets')(io));
-app.use('/layout', require('./routes/databaseRoutes/rest'));
 app.use('/', require('./routes/'));
+defineSockets(io);
```

Comment:
Can socket routes go into routes too?

Reply:
The reason that this is important is because it encapsulates all the routes code in one place. You also want to minimize the point of contact between routes and the server index file.

Reply:
You also want to minimize contact with server/index in general. Most of it was automatically generated and serves as glueing config stuff together.

Reply:
Looking into this more, it seems that it does make some sense to keep defineSockets in the server file.

Reply:
To clarify, what changes should I make here?

Reply:
Use modules

- [ ] Make /server/io.js
- [ ] Put the `io` creation there and `module.exports = io`, i.e. `io` is a singleton
- [ ] server/index requires server/io.js
- [ ] the separate files that handle different socket.io events require server/io.js as well

Reply:
See changes below. I've tagged a spot that still seems iffy.